packages/components/pages/f-contact-preferences/src/components/_tests/ContactPreferences.test.js

```diff
@@ -15,25 +27,209 @@ const i18n = {
     }
 };
 
-const baseProps = {
-    smartGatewayBaseUrl: '/'
+const storeActions = {
+    loadPreferences: jest.fn(),
+    savePreferences: jest.fn(),
+    editPreference: jest.fn()
 };
 
-describe('ContactPreferences', () => {
-    it('should be defined', () => {
-        // Arrange
-        const propsData = {
-            ...baseProps
+const storeState = contactPreferencesViewModel;
```

Comment:
Not for this ticket to be actioned right now, but are we ever likely going to want to have access to getting and setting contact prefs from other parts of the site?

My main reason for asking is, should this be abstracted into more of a service layer, than embedded into contact prefs directly, which seems to hold a lot of the state atm? I would think that some aspects of the contact preferences state may be useful in other pages potentially, so we'd either want to store that state on an application level, or make multiple API calls.

Reply:
We haven't; as yet, in CW needed these values outside of this page but if we do in the future then we can abstract out 👍